packages/good-design/src/hooks/useQueryParam.ts

```diff
@@ -1,4 +1,13 @@
+import { useMemo } from "react";
+
 export const useQueryParam = (param: string): string | null => {
-	const params = new URLSearchParams(window.location.search);
-	return params.get(param)
-}
+  const { search, hash } = window.location;
+
+  return useMemo(() => {
+    // in case of hash we have empty search field
+    const queryString = search || hash?.split("?")[1];
+    const params = new URLSearchParams(queryString);
+
+    return params.get(param);
+  }, [search, hash, param]);
```

Comment:
```suggestion
  }, [param]);
```

non-state vars / props / anything derived from them shouldn't be deps

Reply:
changed